### DISCUSSION TITLE
How to read SVG files?

### DISCUSSION THREAD

**Discussion by SethFalco** [dlemstra/magick-wasm#19] [Q&A]
Sorry, another question. ^-^'

I was tying to read an SVG file, but I get the following exception:

```
Error: NotAuthorized `MVG' @ error/constitute.c/IsCoderAuthorized/422
```

I assume this would normally be circumvented by configuring `policy.xml` on the system with an installed version of ImageMagick. Unfortunately, I'm not sure how this should be handled with Magick.WASM.

Is it possible to read SVGs yet, could I get any advice on how?

**Comment by dlemstra** [comment]
For `SVG` normally the librsvg library would be used but that is not supported. This will require adding support for a lot of libraries. To use the `MVG` delegate the policy will need to be modified but that is not possible yet.

An other option would want to use the `ImageMagick.readFromCanvas` method where the canvas contains an SVG file. That would probably be better than reading it through Magick.WASM.
